lib/fshash/bucket.go

```diff
@@ -58,9 +15,30 @@ func (m Metadata) MarshalBinary() ([]byte, error) {
 */
 type Bucket interface {
 	Record(metadata Metadata, contentHash []byte)
+	Iterator() (rootRecord RecordIterator)
+}
+
+type Record struct {
+	Metadata    Metadata `json:"m"`
+	ContentHash []byte   `json:"h"`
```

Comment:
Actually, no. I originally put that there to indicate that this field *is* serialized, but it turns out I had to do so manually anyway in order to deal with order determinism stuff.  I'm half tempted to leave it there to make the impression and/or document that that's what these fields come out as when serialized, but it is definitely a non-functional ornament.  WDYT, stay or go?